src/Hng.Application/Services/UserService.cs

```diff
@@ -37,5 +53,78 @@ public async Task<IEnumerable<UserDto>> GetAllUsersAsync()
             return _mapper.Map<IEnumerable<UserDto>>(users);
             
         }
+
+
+
+
+
+
+
+
+
+        public async Task<(bool IsSuccess, string ErrorMessage)> IsEmailUniqueAsync(string email)
+        {
+            var existingUser = await _context.Users.FirstOrDefaultAsync(u => u.Email == email);
+            return existingUser == null
+                ? (true, string.Empty)
+                : (false, "Email already exists");
+        }
+
+        public async Task<UserResponseDto> CreateUserAsync(UserSignupDto userSignupDto)
+        {
+            var user = new User
+            {
+                FirstName = userSignupDto.FirstName,
+                LastName = userSignupDto.LastName,
+                Email = userSignupDto.Email,
+                PasswordHash = HashPassword(userSignupDto.Password),
+                CreatedAt = DateTime.UtcNow,
+                UpdatedAt = DateTime.UtcNow
+            };
+
+            _context.Users.Add(user);
+            await _context.SaveChangesAsync();
+
+            return new UserResponseDto
+            {
+                Id = user.Id.ToString(),
+                FirstName = user.FirstName,
+                LastName = user.LastName,
+                Email = user.Email,
+                CreatedAt = user.CreatedAt
+            };
+        }
+
+        public string GenerateJwtToken(UserResponseDto user)
+        {
```

Comment:
Create a token generation service and user that. Because other kinds of token needs to be generated. Call this method `GenerateUserAccessToken`